tests/MongoDB.Driver.Core.Tests/Core/Operations/AsyncCursorTests.cs

```diff
@@ -34,16 +34,161 @@
+                        It.IsAny<Action<IMessageEncoderPostProcessor>>(),
+                        It.IsAny<CommandResponseHandling>(),
+                        It.IsAny<IBsonSerializer<BsonDocument>>(),
+                        It.IsAny<MessageEncoderSettings>(),
+                        It.IsAny<CancellationToken>()),
+                    useKillCursorsCommand ? Times.Once() : Times.Never());
+            }
+            else
+            {
+                subject.Close();
+
+                mockChannelHandle.Verify(
+                    s => s.KillCursors(
+                        It.IsAny<IEnumerable<long>>(),
+                        It.IsAny<MessageEncoderSettings>(),
+                        It.IsAny<CancellationToken>()),
+                    useKillCursorsCommand ? Times.Never() : Times.Once());
+
+                mockChannelHandle.Verify(
+                    s => s.Command(
+                        It.IsAny<ICoreSession>(),
+                        It.IsAny<ReadPreference>(),
+                        It.IsAny<DatabaseNamespace>(),
+                        It.IsAny<BsonDocument>(),
+                        It.IsAny<IEnumerable<Type1CommandMessageSection>>(),
+                        It.IsAny<IElementNameValidator>(),
+                        It.IsAny<BsonDocument>(),
+                        It.IsAny<Action<IMessageEncoderPostProcessor>>(),
+                        It.IsAny<CommandResponseHandling>(),
+                        It.IsAny<IBsonSerializer<BsonDocument>>(),
+                        It.IsAny<MessageEncoderSettings>(),
+                        It.IsAny<CancellationToken>()),
+                    useKillCursorsCommand ? Times.Once() : Times.Never());
+            }
+        }
+
+        [Theory]
+        [ParameterAttributeData]
+        public void Close_should_dispose_cursor_only_once([Values(false, true)]bool async)
+        {
+            int testCursorId = 1;
+            var mockChannelHandle = new Mock<IChannelHandle>();
+            var mockChannelSource = new Mock<IChannelSource>();
+            SetupChannelMocks(mockChannelSource, mockChannelHandle, async, $"{{ 'ok' : true, 'cursorsNotFound' : [], 'cursorsKilled' : [{testCursorId}] }}");
+
+            var subject = CreateSubject(cursorId: 1, channelSource: Optional.Create(mockChannelSource.Object));
+            if (async)
+            {
+                subject.CloseAsync().Wait();
+                mockChannelSource.Verify(c => c.GetChannelAsync(It.IsAny<CancellationToken>()), Times.Once);
+                subject.CloseAsync().Wait();
+                mockChannelSource.Verify(c => c.GetChannelAsync(It.IsAny<CancellationToken>()), Times.Once);
+            }
+            else
+            {
+                subject.Close();
+                mockChannelSource.Verify(c => c.GetChannel(It.IsAny<CancellationToken>()), Times.Once);
+                subject.Close();
+                mockChannelSource.Verify(c => c.GetChannel(It.IsAny<CancellationToken>()), Times.Once);
```

Comment:
I don't understand what these Verify assertions prove.

Reply:
Similar with previous answer, I'm checking how many kill commands were called when we run two Close methods one by one